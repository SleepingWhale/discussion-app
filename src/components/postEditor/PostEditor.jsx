import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './PostEditor.module.css';
import { Button, UserPick } from '../common';

export class PostEditor extends PureComponent {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired
  };

  state = {
    text: ''
  };

  onChange = e => {
    this.setState({
      text: e.target.value
    });
  };

  onSubmit = () => {
    const { text } = this.state;
    const { onSubmit } = this.props;

    onSubmit(text);

    this.setState({
      text: ''
    });
  };

  render() {
    const { text } = this.state;
    const { userPickUrl } = this.props;
    const isSubmitDisabled = text.length === 0;

    return (
      <div className={styles.container}>
        <div className={styles.heading}>
          <UserPick url={userPickUrl} />
          <span className={styles.question}>What&apos;s happening?</span>
        </div>
        <textarea
          className={styles.text}
          maxLength={1000}
          rows={4}
          placeholder="Write something..."
          onChange={this.onChange}
          value={text}
        />
        <div className={styles.buttons}>
          <Button
            onClick={this.onSubmit}
            mode="primary"
            disabled={isSubmitDisabled}
          >
            <span>Post</span>
          </Button>
        </div>
      </div>
    );
  }
}
