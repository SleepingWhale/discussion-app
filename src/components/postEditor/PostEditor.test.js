import React from 'react';
import { mount } from 'enzyme';
import { PostEditor } from './PostEditor';

describe('PostEditor', () => {
  it('should render correctly', () => {
    const onSubmitHandler = jest.fn();
    const component = mount(<PostEditor onSubmit={onSubmitHandler} />);

    expect(component).toMatchSnapshot();
  });

  it('should not submit without content', () => {
    const onSubmitHandler = jest.fn();
    const component = mount(<PostEditor onSubmit={onSubmitHandler} />);
    component.find('button').simulate('click');

    expect(onSubmitHandler).not.toHaveBeenCalled();
  });

  it('should submit with content', () => {
    const content = 'hello world';
    const onSubmitHandler = jest.fn();
    const component = mount(<PostEditor onSubmit={onSubmitHandler} />);
    const input = component.find('textarea').at(0);

    input.instance().value = content;
    input.simulate('change');
    expect(component.state().text).toEqual(content);

    component.find('button').simulate('click');
    expect(onSubmitHandler).toHaveBeenCalledWith(content);
  });
});
