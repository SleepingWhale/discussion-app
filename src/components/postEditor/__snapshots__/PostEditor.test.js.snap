// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostEditor should render correctly 1`] = `
<PostEditor
  onSubmit={[MockFunction]}
>
  <div
    className="container"
  >
    <div
      className="heading"
    >
      <UserPick
        title=""
        url={null}
      >
        <div
          className="placeHolder"
          title=""
        >
          <i
            className="material-icons md-24 md-light"
          >
            sentiment_very_satisfied
          </i>
        </div>
      </UserPick>
      <span
        className="question"
      >
        What's happening?
      </span>
    </div>
    <textarea
      className="text"
      maxLength={1000}
      onChange={[Function]}
      placeholder="Write something..."
      rows={4}
      value=""
    />
    <div
      className="buttons"
    >
      <Button
        disabled={true}
        mode="primary"
        onClick={[Function]}
      >
        <button
          className="base primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span>
            Post
          </span>
        </button>
      </Button>
    </div>
  </div>
</PostEditor>
`;
